In [1]:
import numpy as np
import pandas as pd
import math
import matplotlib.pyplot as plt

In [7]:
data09 = pd.read_excel('DDS9_Data_Extract_with_labels.xlsx', encoding='latin1')
data10 = pd.read_excel('DDS10_Data_Extract_with_labels.xlsx', encoding='latin1')
data11 = pd.read_excel('DDS11_Data_Extract_with_labels.xlsx', encoding='latin1')

In [8]:
print(data09.shape)
print(data10.shape)
print(data11.shape)

(2076, 191)
(2205, 197)
(2131, 198)


By printing the shape of the three year's worth of data we can see two important pieces of take-aways:
1. Each of the years has a similar number of observations. This is good as it means we won't be overreliant on one year's data which may have undue influence on the results.
2. We have a different number of features for each year. Having 1/3 or more of observations automatically not having any data for one of the features is a problem.

In order to account for 2, the first step will be to identify which features are not shared between the years.

In [56]:
cols9, cols10, cols11 = list(data09.columns), list(data10.columns), list(data11.columns)
cols910 = [x for x in cols9 if x in cols10]
shared_cols = [x for x in cols910 if x in cols11]
len(shared_cols)

109

In [36]:
not_cols910 = [x for x in cols9 if x not in cols10]
not_cols109 = [x for x in cols10 if x not in cols9]
not_cols_comb910 = not_cols910 + not_cols109

not_cols_91011 = [x for x in not_cols_comb910 if x not in cols11]
not_cols_11910 = [x for x in cols11 if x not in not_cols_comb910]

not_cols = not_cols_91011 + not_cols_11910
len(not_cols)

204

In [45]:
cols1011 = [x for x in cols10 if x in cols11]
len(cols1011)

182

Here we can see that there are 109 shared columns between all three years and 204 columns that don't share the same name. However, given the similarity in size of years 10 and 11, we assumed that they might have greater similarity. It turns out they did, with 182 columns sharing the exact same name. Therefore, we decided to merge these two years and ignore the data from year 9. 

Before we merged the sets, we had to adjust the names of our target variable in order to make that a shared element. This would then bring a total of 183 shared columns.

In [52]:
data10['target_willing'] = data10.iloc[:, -3]
data11['target_willing'] = data10.iloc[:, -3]

The final piece before we merge the datasets is to add a column indicating the year in case that ends up proving relevant in the analysis.

In [53]:
data10['year'] = 10
data11['year'] = 11

In [58]:
merge_cols = [x for x in cols10 if x in cols11]

merge10 = data10[merge_cols]
merge11 = data11[merge_cols]

df = pd.concat([merge10, merge11])

In [60]:
df.head()

record - Record number  Q1r1 - To begin, what is your age?  \
0                       7                                  31   
1                       4                                  30   
2                       8                                  61   
3                       3                                  68   
4                      15                                  50   

  Q4 - What is your gender? age - you are...  \
0                    Female            30-46   
1                    Female            30-46   
2                      Male            47-65   
3                    Female      66 or older   
4                    Female            47-65   

  Q2 - In which state do you currently reside? region - Region  \
0                                     Illinois         Midwest   
1                                     Arkansas           South   
2                                      Alabama           South   
3                                     New York       Northeast   
4                                         Iowa         Midwest   

  QNEW3 - What is your employment status?  \
0                              Unemployed   
1                              Unemployed   
2                                 Retired   
3                                 Retired   
4         Employed full-time or part-time   

  Q5 - Which category best describes your ethnicity?  \
0                  White or Caucasian (Non-Hispanic)   
1                  White or Caucasian (Non-Hispanic)   
2                  White or Caucasian (Non-Hispanic)   
3                  White or Caucasian (Non-Hispanic)   
4                  White or Caucasian (Non-Hispanic)   

  QNEW1 - Do you have children living in your home (excluding yourself if you are under 18)?  \
0                                                Yes                                           
1                                                Yes                                           
2                                                 No                                           
3                                                 No                                           
4                                                 No                                           

  QNEW2 - How old are the children in your home?-0-4 years  ...  \
0                                                 No        ...   
1                                                Yes        ...   
2                                                NaN        ...   
3                                                NaN        ...   
4                                                NaN        ...   

  Q39rNEW1 - I would rather pay for sports information online in exchange for not being exposed to advertisements. - Using the scale below, please indicate how much you agree or disagree with the following statements. If the question does not apply to you, c  \
0                                     Agree somewhat                                                                                                                                                                                                                 
1                                  Disagree strongly                                                                                                                                                                                                                 
2                                  Disagree strongly                                                                                                                                                                                                                 
3               N/A; I do not have a basis to answer                                                                                                                                                                                                                 
4                           

In [59]:
df.shape

(4336, 184)